app/services/intercom_service.rb

```diff
@@ -0,0 +1,113 @@
+require 'intercom'
+
+class IntercomService
+  def self.create_intercom_message_from_email(incoming_email, fw_from_hub: false)
+    email = incoming_email.client&.intake&.email_address || incoming_email.sender
+    body = incoming_email.body
+    contact_id_from_email = contact_id_from_email(email)
+    contact_id = contact_id_from_email.present? ? contact_id_from_email : create_intercom_contact(incoming_email).id
+
+    if contact_id_from_email.present? && most_recent_conversation(contact_id).present?
+      reply_to_existing_intercom_thread(contact_id, body)
+    else
+      create_new_intercom_thread(contact_id, body)
+      send_handoff_email(incoming_email.client) if fw_from_hub
+    end
+  end
+
+  def self.create_intercom_message_from_sms(incoming_sms, fw_from_hub: false)
```

Comment:
In what cases are we or would we pass fw_from_hub as false? Also, does fw_from_hub being true only affect the sending of the handoff email in the case that it is necessary? in that case, may make sense to change the option to something like `handoff` or `inform_of_handoff`

Reply:
Yeah I definitely struggled to name this one well but I like `inform_of_handoff`! I was looking at future scenarios where we might want to create intercom messages that don't come from the hub and thought that having this option would make that easier but maybe that was too pre-emptive of me

Reply:
oh i see! I think if we're never executing the code with fw_from_hub as false in the codebase currently, we can probably remove keyword arg altogether for now and just allow it to inform of handoff any time the "else" scenario runs, and then when the case comes up that we sometimes _dont_ want to send the handoff, we can implement this switch.